src/dbus_api.rs

```diff
@@ -286,7 +284,60 @@ fn destroy_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 }
 
 fn list_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
-    Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    if iter.arg_type() != 0 {
```

Comment:
We could start complaining if there are too many args, but that seems like overkill. In any case, we should do it consistently, if we decide to.